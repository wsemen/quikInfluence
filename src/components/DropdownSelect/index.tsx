import { FormControl, FormErrorMessage, FormLabel, FormLabelProps, Select, SelectProps } from '@chakra-ui/react';
import React from 'react';
import quikColorConstants from 'utils/constants/colorConstants';

type DropdownSelectOption = {
  label: string;
  value: string;
};

interface DropdownSelectProp {
  size?: string;
  label?: string;
  options: DropdownSelectOption[];
  onChange: (event: React.ChangeEvent<HTMLSelectElement>) => void;
  inputId?: string;
  labelProps?: FormLabelProps;
  selectProps?: SelectProps;
  error?: string
}

const DropdownSelect = ({
  size = 'lg',
  label,
  options,
  onChange,
  inputId='',
  labelProps,
  selectProps,
  error
}: DropdownSelectProp) => {
  return (
    <FormControl isInvalid={!!error}>
      {
        !!label &&
          <FormLabel
            fontSize='1.6rem'
            color={quikColorConstants.black}
            htmlFor={inputId}
            {...labelProps}>
            {label}
          </FormLabel>
      }
      <Select
        onChange={onChange}
        size={size}
        border='1px solid #D5D5DC'
        borderRadius='xl' 
        id={inputId}
        placeholder={`Select ${label || '---'}`}
        data-test-id="select-component"
        {...selectProps}
      >
        {options.map(option => {
          return (
            <option data-testid="select-option" key={option.value} value={option.value}>
              {option.label}
            </option>
          );
        })}
      </Select>
      {
        error &&
        <FormErrorMessage 
          fontSize='xl'>
            {error}
        </FormErrorMessage>
      }
    </FormControl>
  );
};

export default DropdownSelect;
