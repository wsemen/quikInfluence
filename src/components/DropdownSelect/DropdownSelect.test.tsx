import { render } from '@testing-library/react';
import DropdownSelect from '.';

it('renders DropdownSelect component', async () => {
  const mockedOnChange = jest.fn();
  const { container } = render(
    <DropdownSelect
      options={[
        { value: '1', label: 'Option 1' },
        { value: '2', label: 'Option 2' },
        { value: '3', label: 'Option 3' },
      ]}
      onChange={mockedOnChange}
    />
  );
  expect(container).toMatchSnapshot();
});
