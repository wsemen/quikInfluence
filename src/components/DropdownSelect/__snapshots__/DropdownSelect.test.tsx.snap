// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders DropdownSelect component 1`] = `
<div>
  <div
    class="chakra-form-control css-0"
    role="group"
  >
    <div
      class="chakra-select__wrapper css-42b2qy"
    >
      <select
        class="chakra-select css-l65hh7"
        data-test-id="select-component"
        id=""
      >
        <option
          value=""
        >
          Select ---
        </option>
        <option
          data-testid="select-option"
          value="1"
        >
          Option 1
        </option>
        <option
          data-testid="select-option"
          value="2"
        >
          Option 2
        </option>
        <option
          data-testid="select-option"
          value="3"
        >
          Option 3
        </option>
      </select>
      <div
        class="chakra-select__icon-wrapper css-162mkon"
      >
        <svg
          aria-hidden="true"
          class="chakra-select__icon"
          focusable="false"
          role="presentation"
          style="width: 1em; height: 1em; color: currentColor;"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
            fill="currentColor"
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
